### ISSUE TITLE
IKEA / Zigbee devices uncontrollable 1.5.5

### ISSUE THREAD

**Issue by dbonsaksen** [athombv/homey#1956]
**Metadata**

* **Homey Version**: 1.5.5
* **Homey Log Code**: B6942BBAAF
* **Smartphone Platform**: IOS
* **Smartphone App Version**:

**If related to a Homey App:**
* **Homey App Version**: IKEA / KaKu Zigbee
* **Homey App Log Code**:
* **Affected Devices (preferably product codes)**:

**What did you try to achieve?**
I updated to 1.5.5 today as this was supposed to fix the issue with zigbee devices being uncontrollable, it does absolutely not fix it. It actually made it worse, as it made more bulbs uncontrollable. So I decided to reset my entire zigbee network, and pairing all devices again. I was able to pair every bulb and was able to control them for 5 minutes, then we were back to scratch.

Im trying to get these Gu10 dimmable bulbs to turn on by the help of an motion sensor. This is impossible as it might take up to 10-15 secounds before it turns on, if they turn on at all.

If I try to control them with the smartphone app I get the message: "Something went wrong addressing device XXXXX"

I have tried PTP, resetting my zigbee network, reincluding devices and it does not solve with anything I can do.

**What did you expect as result?**
Getting fixed from the bug in 1.5.4

**What was the result?**
Not being able to control my any of my zigbee devices.

**Any other remarks**
Please do not close this and say it has been solved, because it is definitely not solved!

**Comment by johane77** [comment]
I have the exact problem on 1.5.5...

**Comment by johane77** [comment]
If I turn off/on each lamp, some will give same message, but after a few tries it will most of the time react. Its totally useless to group lamps and turn on/off. Zigbee issue is deffinitly not solved.

**Comment by johane77** [comment]
Please fix this soon. I'm so frustrated. I thought Zigbee would have the advantage of the handshaking, but it is really worse than 433mhz at the moment.

**Comment by RobinBol** [comment]
@dbonsaksen thanks for creating this issue.

First, a Flow with a motion sensor may indeed have a delay since the motion has to be detected first. Since you did not mention which device/protocol you are using for this I can not comment on the delay.

Second, ZigBee devices are not responding to commands, but can you describe how we can reproduce your issue? What does your network look like? Is it large? Many devices? Did you move Homey after pairing, etc. Every bit of information would be useful. From the logs I can not see anything really strange, it does look like you have a lot of devices that are trying to connect with Homey all at once. Could you maybe try to reboot Homey with most zigbee devices turned off, and then turn them on in small batches over lets say 10 minutes?

**Comment by johane77** [comment]
I have approx 45 Ikea Trådfri,  RGB bulbs, dimmable white and spots. Turn on lamps works almost every time. But turning off lamps only works if I go to each lamp in app and turn of. Creating a group in device groups app works really bad, different lamps every time failing to turn off. This is clearly worse in 1.5.5. I have android phone and both apps installed on phone. Both makes also issue sometime that it say that homey is offline, this is happening in middle of turn off sequence. Maybe you need to slow down sequence if many lamps is turned of at the same time? Please help! What do you need to solve this?

**Comment by johane77** [comment]
This just happened! When turning off lamps! 

![screenshot_20171120-191257](https://user-images.githubusercontent.com/33071935/33034099-0508c6fc-ce27-11e7-90e7-99a8e0bdad6e.png)
![screenshot_20171120-191259](https://user-images.githubusercontent.com/33071935/33034121-116b11b6-ce27-11e7-95e4-d681fd051b3f.png)

**Comment by johane77** [comment]
30 of 45 lamps failed..

**Comment by RobinBol** [comment]
Could you send a system log?

**Comment by johane77** [comment]
![20171120_203149](https://user-images.githubusercontent.com/33071935/33037475-f47ab1f0-ce31-11e7-8655-1d1a091e96d1.jpg)

**Comment by johane77** [comment]
Since it failed its extremely slow, reconnecting over and over again. Log is sent so I will restart it and send a new log. Could that help?

**Comment by johane77** [comment]
![20171120_204026](https://user-images.githubusercontent.com/33071935/33037854-121b5416-ce33-11e7-91fd-88403d405855.jpg)

**Comment by johane77** [comment]
Any fault found?

**Comment by johane77** [comment]
Feedback RobinBol?

**Comment by RobinBol** [comment]
We are looking into this issue, thanks for the logs. If I need more information I will give you an update here.

**Comment by johane77** [comment]
RobinBol, just curious if you found any issues?

**Comment by RobinBol** [comment]
@dbonsaksen since you opened this issue could you take a look at my initial comment and provide some feedback?

@johane77 not necessarily from your logs, but we are aware of this issue and working to improve the situation. If you have other issues (your Homey going offline), please create separate issues for that so that we can keep the discussion on track here.

**Comment by dbonsaksen** [comment]
@RobinBol Sorry I missed your response here.

I have 6 GU10 bulbs in my bathroom roof which is supposed to be turned on by a sensor. But the problem is that most of them does not respond to using the app (cant adress them).

They are aprox 3 meters from Homey through 2 wooden walls, and the homey hasnt been moved. It does not matter if I use them in a flow or not, they are not responsive with the app either way.

**Comment by RobinBol** [comment]
@dbonsaksen thanks for your response! Could you verify that if you remove the power from the GU10 bulbs for 30 seconds, then restore the power and wait 5 minutes, that the bulbs become controllable via de Homey smartphone app again? This might help pointing us in the right direction.

**Comment by dbonsaksen** [comment]
Last time I tried that, they became controllable for a short period of time, then I was back to scratch.

Exact same thing is happening if I remove them entierly, they work once I reinclude them but stops working after a little while.

**Comment by VincentWRX** [comment]
2 days ago i woke up with 1 of my 3 lights turned on. While i was 100% sure they were off. It sometimes looks like a really big delay. Sometimes 2 of the 3 turn on, sometimes 1 turns white and the other one yellow.  Got the 3 of them in a group. but also using the app is not working correctly. colour is not changing, dim sometimes work. So hope this gets solved quick as i don't want to swap back to the hue bridge again.

**Comment by ljsvv** [comment]
I think something has to do with the Tradfri's not creating a Mesh from bulb to bulb but always choose a direct path to Homey. All my trader's are directly connected to Homey and I also experience that not always all bulbs go on/off...

Below are 2 screenshots of my Zigbee devices, not 1 single Bulb is connected trough a Mesh...

<img width="1271" alt="route1" src="https://user-images.githubusercontent.com/29465686/34015847-1277a4d2-e120-11e7-9920-504fab012c0c.png">

<img width="1271" alt="route2" src="https://user-images.githubusercontent.com/29465686/34015907-45494082-e120-11e7-913d-03fddfae7cb7.png">

**Comment by johane77** [comment]
Indeed nothing got better. Even harder to add Gu10 Turnable spots.

What is going on?

**Comment by dbonsaksen** [comment]
Update on 1.5.6.RC7: I re-included all the IKEA devices, and I am now able to control them. BUT, it's not working 100% of the times I try to turn them on/off, I would say about 90% and they might be very slow.

It's also a problem that if you turn them all on at once or very quickly after another they will have a very high chance of not responding. So I basically have to turn them on one by one with a secound between.

So they have sort of worked for a couple of days now, but unfortunately very far from reliable.

**Comment by johane77** [comment]
Exactly, could have been me writing! Exakt same issue. Controllable but needs to be set one by one.



Mvh Johan Eriksson

Tel. 0723356500


-------- Original message --------

**Comment by johane77** [comment]
Any way to solve this? Getting tired of having an angry girlfriend... Every time I turn on or off the ikea trådfri lamps and 433mhz sockets that are grouped in Device groups, different lamps decide to not communicate. They are possible to to turn off or on manually directly after, but why are they failing, and why is it different lamps every time? 

Please, tell me what to do, should I get rid of Homey, or the Ikea trådfri, or is this just software issues that takes time to fix on your side?

Its frustrating that new functions are developed, but standard functions like turn on/off is failing. 

Please advise what I can do???

Regards Johan Eriksson

**Comment by ljsvv** [comment]
Agree, Not even a Message that the issue is under investigation.... I understand that Homey has little team that develops. But we paid money for a product that is really not market ready imo!!!

**Comment by RobinBol** [comment]
@ljsvv my last response was 10 days ago, and I mentioned that the ZigBee issues are something we are working on. We take them seriously, but unfortunately they take time to resolve.

@johane77 we are working to improve the ZigBee reliability, I understand your frustration but we believe that in time the situation will be improved!

**Comment by johane77** [comment]
Thank you for responding! I do feel that more information about the timeframe for issues should be communicated. Zigbee has been an issue from beginning, and If I know it's doable for your team I will wait. Otherwise I will leave Homey.

**Comment by ljsvv** [comment]
@RobinBol thanks.

I understand things are different and time is flying by with an 300+ issue counting GH.

Please let me know If  you need some additional data, as you can see I have quite a lot of Zigbee Grid connected stuff, (None of them make a mesh underneath).

**Comment by ljsvv** [comment]
Please also look at the no mesh between the Tradfri’s underneath issue. I posted a picture of my “mesh” above. For example the bulb Washok Rechts has 0% link quality still directly connected to Homey and doesn’t always switch on/off. I do get a no ack message when trying to switch this bulb on and off. I’m afraid the Ikea Tradfri’s do not use the Zigbee mesh underneath.....

**Comment by johane77** [comment]
Now I have left Homey at store. If you solve issues with Zigbee I will be a customer again, but you seem to have huge problems to solve our issues. Maybe because of bad hardware. Who knows except the Athom team. Good luck and hopefully this will be the best product in the future!

**Comment by ljsvv** [comment]
Situation got already a little bit better in 1.5.6 RC8. Still had some lights that did not switch when switching multiple bulbs at once.

**Comment by RobinBol** [comment]
The error tells me that the device was reached with the command, but the device itself did not respond.

**Comment by johane77** [comment]
Had same errors with my Trådfri lamps. None with Hue Hub...

**Comment by ljsvv** [comment]
@RobinBol True, but like said, even powered down the bulb and repowered didn;t fix it. Only rebooting Homey made the bulb react immediately again after all apps where loaded!!!

**Comment by ljsvv** [comment]
Sorry! Moved my lights that where the most problematic to the Philips Hue Bridge and the same lights work like a charm now!

**Comment by johane77** [comment]
Yes, Homey needs to adress this and learn from Philips, until then it's not an option for me. If I import ant try to turn on or off with homey and lamps connected to hue Hub then it's problem again. The fault is clearly at the Homey side..

**Comment by bvdbos** [comment]
Really guys, this is not a bug, it's a feature-request for groups. I suggest closing this bug in favor of #2025 or maybe #1859 ? And file a new bug for https://github.com/athombv/homey/issues/1956#issuecomment-356105533 ?

**Comment by ljsvv** [comment]
Please read my post before judging this is a request for groups. Even if I switch on bulbs with a delay of 1 seconds bulbs do not turn on sometimes

**Comment by johane77** [comment]
Exactly ljsvv!

**Comment by VincentWRX** [comment]
When will this be fixed? i only got 3 lightify leds, but i can't turn them off. Even not in the app. Sometimes 1 goes on, or 2 go on.. and 1 goes off.. Thinking to connect them again to my philips hue.. But this is really not working.

**Comment by bvdbos** [comment]
see also #2325

**Comment by EspenHalvorsen** [comment]
My 14 IKEA GU10 spots are making my wife mad. They dim and turn on/off in a randomly pattern. 
And they dont support the IKEA remotes, so all we have is the app.

**Comment by GitHubLuigi** [comment]
@RobinBol, you wrote this almost a year ago.

“@ljsvv my last response was 10 days ago, and I mentioned that the ZigBee issues are something we are working on. We take them seriously, but unfortunately they take time to resolve.

@johane77 we are working to improve the ZigBee reliability, I understand your frustration but we believe that in time the situation will be improved!”

Since the issues still excist in the same way they did back then and your team has been working on this for almost a year without succes, is it reasonable to conclude the Zigbee issues are unfixable? 
I would like to know because that could possibly mean I should move on to be using another system instead, since Im using mainly Zigbee devices.

**Comment by ekouters** [comment]
Today I read up on Athom's monthly [Behind the Magic blog](https://blog.athom.com/category/behind-the-magic/), and noticed in the [July edition](https://blog.athom.com/behind-magic-july-2018/) that the engineer assigned to Zigbee can no longer work on this anymore:

> Zigbee
> We planned to rewrite pieces of our Zigbee stack to improve reliability, but unfortunately due to circumstances with the engineer assigned on the project not being able to anymore this has been delayed a bit. We’re always on the lookout for talent.

This could explain the lack of progress on this issue.
